### ISSUE TITLE
Caching of hydra:ApiDocumentation and Entrypoint

### ISSUE THREAD

**Issue by lazka** [api-platform/core#4074]
For using the json-ld/hydra API I have to check https://demo.api-platform.com/docs.jsonld and https://demo.api-platform.com/.jsonld to figure out the types and operations which means one extra request before the first API call. These requests aren't cached, and slow things down.

I wonder if this couldn't be cached with etags since the content is static anyway. Am I missing something maybe?

**Comment by dunglas** [comment]
You're right, we must set etags and basic HTTP cache header on these endpoints, this could improve the performance of the admin a lot.

PR welcome!!

**Comment by J3m5** [comment]
It would be really nice to have this.
Not only the admin but all the clients that use the api doc parser.
For us, the docs are about 280 KB in size and takes 1.3 s to fetch.

**Comment by abderrahimghazali** [comment]
Opened #7929 to address this. The PR adds an event listener that sets a strong ETag (md5 of the body) plus `Cache-Control: public, max-age=0, must-revalidate` on `api_doc` and `api_entrypoint` responses, and lets `Response::isNotModified()` short-circuit to a 304 when the client sends a matching `If-None-Match`. Feedback welcome!
